networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/AttestationTopicSubscriber.java

```diff
@@ -36,25 +40,62 @@ public AttestationTopicSubscriber(final Eth2Network eth2Network) {
   public synchronized void subscribeToCommitteeForAggregation(
       final int committeeIndex, final UnsignedLong aggregationSlot) {
     final int subnetId = committeeIndexToSubnetId(committeeIndex);
-    final UnsignedLong currentUnsubscriptionSlot =
-        unsubscriptionSlotBySubnetId.getOrDefault(subnetId, ZERO);
+    final UnsignedLong currentUnsubscriptionSlot = allSubscriptions.getOrDefault(subnetId, ZERO);
     if (currentUnsubscriptionSlot.equals(ZERO)) {
       eth2Network.subscribeToAttestationSubnetId(subnetId);
     }
-    unsubscriptionSlotBySubnetId.put(subnetId, max(currentUnsubscriptionSlot, aggregationSlot));
+    allSubscriptions.put(subnetId, max(currentUnsubscriptionSlot, aggregationSlot));
+  }
+
+  public synchronized void subscribeToPersistentSubnets(
+      final Set<SubnetSubscription> newSubscriptions) {
+    boolean shouldUpdateENR = false;
+
+    for (SubnetSubscription subnetSubscription : newSubscriptions) {
+      int subnetId = subnetSubscription.getSubnetId();
+      if (!persistentSubnetIdSet.contains(subnetId)) {
+        persistentSubnetIdSet.add(subnetId);
+        shouldUpdateENR = true;
+      }
+
+      UnsignedLong existingUnsubscriptionSlot =
+          allSubscriptions.computeIfAbsent(
+              subnetId,
+              (key) -> {
+                eth2Network.subscribeToAttestationSubnetId(subnetId);
+                return ZERO;
+              });
+
+      allSubscriptions.put(
+          subnetId, max(existingUnsubscriptionSlot, subnetSubscription.getUnsubscriptionSlot()));
```

Comment:
(optional) You could do everything in one compute operation:
```suggestion
          allSubscriptions.compute(
              subnetId,
              (key, currentValue) -> {
                final UnsignedLong unsubscribeSlot = subnetSubscription.getUnsubscriptionSlot();
                if (currentValue == null) {
                    eth2Network.subscribeToAttestationSubnetId(subnetId);
                    return unsubscribeSlot;
                } else {
                    return max(existingUnsubscriptionSlot, unsubscribeSlot);
                }
              });
```